lib/filterx/object-datetime.c

```diff
@@ -205,6 +205,38 @@ _repr(FilterXObject *s, GString *repr)
   return datetime_repr(&ut, repr);
 }
 
+static FilterXObject *
+_add(FilterXObject *lhs_object, FilterXObject *rhs_object)
+{
+  if (!filterx_object_is_type(lhs_object, &FILTERX_TYPE_NAME(datetime)))
+    goto error;
```

Comment:
If this is not `datetime`, it is a programming error, so this should be an assert or we could just simply remove it. This applies to the rest of the types' `add()`, too.